umbral/pre.py

```diff
@@ -286,39 +287,44 @@ def reencrypt(kfrag: KFrag, capsule: Capsule, params: UmbralParameters=None,
 def _prove_correctness(cfrag: CapsuleFrag, kfrag: KFrag, capsule: Capsule, 
                        metadata: bytes=None, params: UmbralParameters=None
                       ) -> CorrectnessProof:
-    params = params if params is not None else default_params()
+    t = CurveBN.gen_rand(params.curve)
+    rk = kfrag._bn_key
 
-    e1 = cfrag._point_e1
-    v1 = cfrag._point_v1
+    params = params if params is not None else default_params()
 
```

Comment:
`params` initialization should be done in the first line. I would also prefer to keep `t = CurveBN.gen_rand(params.curve)` just in the previous line to its first use `e2 = t * e`, but that's a matter of personal preference.

Reply:
Right - `params` was a goof here.  

On `t`, I had the same inclination, but then that makes it impossible to do a line-for-line match with the verify logic.